src/mgvalue.c

```diff
@@ -1639,6 +1640,48 @@ mg_path *mg_path_make(uint32_t node_count, mg_node **nodes,
   return path;
 }
 
+mg_date *mg_date_make(int64_t days) {
+  mg_date *date = mg_date_alloc(&mg_system_allocator);
+  if (!date) {
+    return NULL;
+  }
+  date->days = days;
+  return date;
+}
+
+mg_local_time *mg_local_time_make(int64_t nanoseconds) {
+  mg_local_time *lt = mg_local_time_alloc(&mg_system_allocator);
+  if (!lt) {
+    return NULL;
+  }
+  lt->nanoseconds = nanoseconds;
+  return lt;
+}
+
+mg_local_date_time *mg_local_date_time_make(int64_t seconds,
+                                            int64_t nanoseconds) {
+  mg_local_date_time *ldt = mg_local_date_time_alloc(&mg_system_allocator);
+  if (!ldt) {
+    return NULL;
+  }
+  ldt->seconds = seconds;
+  ldt->nanoseconds = nanoseconds;
+  return ldt;
+}
+
+mg_duration *mg_duration_make(int64_t months, int64_t days, int64_t seconds,
+                              int64_t nanoseconds) {
+  mg_duration *dur = mg_duration_alloc(&mg_system_allocator);
+  if (!dur) {
+    return NULL;
+  }
+  dur->months = months;
+  dur->days = days;
+  dur->seconds = seconds;
+  dur->nanoseconds = nanoseconds;
```

Comment:
Aren't we also using microseconds, or are those encompassed in the nanoseconds?

Reply:
Good question! Anything below seconds is converted into nanoseconds and sent over the wire. The caller of this function must make this conversion.

Reply:
everything here is the implementation of the PackStream for temporal types:
https://7687.org/packstream/packstream-specification-1.html#localdatetime---structure